app/services/facets/entry.rb

```diff
@@ -0,0 +1,53 @@
+module Facets
+  class Entry
+    ATTRIBUTES = [:name, :model, :helper, :extension, :tabs]
+    attr_accessor(*ATTRIBUTES)
+
+    ATTRIBUTES.each do |attr|
+      define_method "#{attr}_class".to_sym do
+        sym = self.send(attr)
+        to_class(sym)
+      end
+    end
+
+    def initialize(facet_name, facet_model)
```

Comment:
Invert the order of arguments here and find the facet_name from the model, it would fit most cases